x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/ElasticInferenceServiceFeature.java

```diff
@@ -15,6 +15,8 @@
  */
 public class ElasticInferenceServiceFeature {
 
+    @Deprecated
     public static final FeatureFlag ELASTIC_INFERENCE_SERVICE_FEATURE_FLAG = new FeatureFlag("eis");
```

Comment:
Maybe we could rename `ELASTIC_INFERENCE_SERVICE_FEATURE_FLAG` to `DEPRECATED_INFERENCE_SERVICE_FEATURE_FLAG`. Than it's also already clear from the constant name, which one to use.